UIAutomationTests/TestsLogging.cs

```diff
@@ -0,0 +1,126 @@
+            //it keeps writing to it as well, not sure why ..
+            //we pre-emptively kill this process if it's still running before we run each of these tests 
+            //since it holds the log file hostage and could also mess with our test results.
+            string processName = "Microsoft.Mashup.Container.NetFX45";
+
+            var processes = Process.GetProcessesByName(processName);
+            foreach ( var process in processes) {
+                process.Kill();
+            }
+
+            Task.Delay(TimeSpan.FromSeconds(2)).Wait();
+
+            var logFilePath = Utilities.GetConfigurationValue("odbcLogFilePath");
+            if (File.Exists(logFilePath))
+                File.Delete(logFilePath);
+        }
+
+        private string FillInLogFilePath(string MQueryExpression)
+        {
+            var logFilePath = Utilities.GetConfigurationValue("odbcLogFilePath");
+            var newMQueryExpression = MQueryExpression.Replace("{absoluteLogPath}", logFilePath);
+            return newMQueryExpression;
+        }
+        private void AssertLogActivityAfterTs(DateTime startTestTimeStamp)
+        {
+            var logFilePath = Utilities.GetConfigurationValue("odbcLogFilePath");
+            Task.Delay(TimeSpan.FromSeconds(timeToWaitForOdbcDriverToStartLoggingInSeconds)).Wait();
+
+            var lastWrittenToLogFileDt = File.GetLastWriteTime(logFilePath);
+            var testCheckTime = DateTime.Now;
+            output.WriteLine("Log file path: " + logFilePath);
+            output.WriteLine($@"check at {testCheckTime}, test start time : {startTestTimeStamp.ToLongTimeString()} ,last time modified: {lastWrittenToLogFileDt.ToLongTimeString()}");
+            Assert.True(lastWrittenToLogFileDt > startTestTimeStamp, $@"Nothing's been logged, check at {testCheckTime}, test start time : {startTestTimeStamp.ToLongTimeString()} ,last time modified: {lastWrittenToLogFileDt.ToLongTimeString()}");
+        }
+                
+        private void AssertNoLogActivityAfterTs(DateTime startTestTimeStamp)
+        {
+
+            var logFilePath = Utilities.GetConfigurationValue("odbcLogFilePath");
+            Task.Delay(TimeSpan.FromSeconds(timeToWaitForOdbcDriverToStartLoggingInSeconds)).Wait();
+
+            var lastWrittenToLogFileDt = File.GetLastWriteTime(logFilePath);
+            var testCheckTime = DateTime.Now;
+            output.WriteLine("Log file path: " + logFilePath);
+            output.WriteLine($@"check at {testCheckTime}, test start time : {startTestTimeStamp.ToLongTimeString()} ,last time modified: {lastWrittenToLogFileDt.ToLongTimeString()}");
+            Assert.False(lastWrittenToLogFileDt > startTestTimeStamp, "Something's been logged");
+        }
+        [Fact]
+        public void LogPathArgumentFilledIn()
+        {
+            var startTestTimeStamp = DateTime.Now;
+
+            string MQueryExpression = File.ReadAllText("QueryPqFiles/LogPathArgument.query.pq");
```

Comment:
I think would be nice to see the test setup in the tests instead of having separated files for it.
Maybe it's possible to copy the contents of `QueryPqFiles/LogPathArgument.query.pq` here.

Same with all the other tests. Maybe for a future PR

Reply:
I thought about this : One of the big negatives of VS/C# is that multiline strings don't format very well in the editor.
There's also some special characters that would make this even more ugly.